### PR TITLE
Node controller.

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

Add docs on the node controller.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Add docs on the node controller.
```

### CODE REVIEW

File: v2.6/usage/decommissioning-a-node.md [hunk 1]
```diff
@@ -11,9 +11,15 @@ redirect_from: latest/usage/decommissioning-a-node
 - You are receiving an error about an IP address already in use.
 - Hosts are regularly added and removed from your cluster.
 
+### Why you might not be interested in this guide
+
+- You are running Calico in Kubernetes-Datastore Mode, which automatically
+handles node deletion.
+- You are running the Calico Policy Controller v1.0.1+, which automatically handles node deletion.
```
Comment (caseydavenport) [inline v2.6/usage/decommissioning-a-node.md]: I think we should re-phrase this in a positive way. Rather than "Why you're not interested in this" it should be "When should you read this document"

The reasons shouldn't talk about the policy controller (it's the node controller that does this) and it really should talk more in terms of user-facing components, so "kube-controllers" is really appropriate.

Users will also need to follow this if not using self-hosted or another approach which populates the k8s node references.

File: v2.6/usage/decommissioning-a-node.md [hunk 2]
```diff
@@ -11,9 +11,15 @@ redirect_from: latest/usage/decommissioning-a-node
...
+
 ### Purpose of this page
 
-Provide guidance on how to remove a host that is part of a Calico cluster
+Provide guidance on how to manually remove a host that is part of a Calico cluster
```
Comment (caseydavenport) [inline v2.6/usage/decommissioning-a-node.md]: I'd undo this change - the information in this doc can still apply to homespun automated processes.

**Comment by emanic** [comment]
This will be nice to have in v3.0 and master as well.

**Comment by emanic** [comment]
@ozdanborne Could remove the decommissioning a node part (addressed in separate PR) and perhaps @caseydavenport might approve this and we could then merge it, potentially.

File: v2.6/usage/decommissioning-a-node.md [hunk 3]
```diff
@@ -10,6 +10,9 @@ redirect_from: latest/usage/decommissioning-a-node
 - You are renaming a Node.
 - You are receiving an error about an IP address already in use.
 - Hosts are regularly added and removed from your cluster.
+- You are not running Calico in Kubernetes-Datastore Mode, which automatically
```
Comment (caseydavenport) [inline v2.6/usage/decommissioning-a-node.md]: Let's remove the changes in this file in favor of the other PR.

**Comment by ozdanborne** [comment]
Updated as per requested changes. Test is failing due to some external link in v2.5. Hoping we can ignore for this PR and merge
